cpp/src/gandiva/engine.cc

```diff
@@ -86,6 +88,13 @@
 #include <llvm/Transforms/Utils.h>
 #include <llvm/Transforms/Vectorize.h>
 
+// JITLink is available in LLVM 9+
+// but the `InProcessMemoryManager::Create` API was added since LLVM 14
+#if LLVM_VERSION_MAJOR >= 14 && !defined(_WIN32) && !defined(_WIN64)
```

Comment:
Do we need `!defined(_WIN64)` here?
(Does this work on 64bit Windows?)

Reply:
I initially didn't include this, but without including `!defined(_WIN64)` makes several CI checks on Windows failed, sorry I didn't kept the specific CI job links for that but I remembered the CI reported something like "incorrect object file format". 

`JITLink` does support Windows/COFF format [1][2], but it is fairly new feature, probably landed in LLVM 15 (released in 2022-09), and since I tried to make the `JITLink` related change minimum, and I don't have Windows env to cover different LLVM versions' impact, so I simply skip Windows in this case to make the version check simple.

[1] NATIVE WINDOWS JITING IN LLVM, https://llvm.org/devmtg/2022-11/slides/Tutorial2-JITLink.pdf
[2] 2022 LLVM Dev Mtg: JITLink: Native Windows JITing in LLVM, https://www.youtube.com/watch?v=UwHgCqQ2DDA

Reply:
Hmm. I think that we can skip all Windows with just `!defined(_WIN32)`. Because `_WIN32` is defined on 64bit Windows too.

https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170

> `_WIN32` Defined as 1 when the compilation target is 32-bit ARM, 64-bit ARM, x86, or x64. Otherwise, undefined.